lib/suse-cloud-upgrade-4-to-5-pre

```diff
@@ -206,7 +206,27 @@ for node in $allocated_nodes; do
 done
 
 # Make sure all openstack services are down on all nodes before we stop rabbitmq
-sleep 10
+services_stopped=""
+count=0
+while [ -z "$services_stopped" -a "$count" -lt 100 ]; do
+  sleep 1
+  services_stopped="yes"
+  for node in $allocated_nodes; do
+    if ssh "$node" 'shopt -s nullglob
+      for i in /etc/init.d/openstack-* ; do
+        if $i status >/dev/null ; then
+          echo "$i still running at `hostname`"; exit 0
+        fi
+      done
+      exit 1'; then
+      services_stopped=""
```

Comment:
Reading the code again, I know how to be less confused about this. Is it more readable to do:

```sh
while [ "x$servise_stopped" != "xyes" -a ...]; do
  services_stopped="yes"
  for ...
    if ...
      services_stopped="no"
    fi
  done
done
```